rivers/books_drupal.json

```diff
@@ -0,0 +1,131 @@
+      "sys_id": {"remote_field": "prep_id"},
+      "sys_tags": {"remote_field": "prep_sys_tags"},
+      "sys_title": {"remote_field": "prep_sys_title"},
+      "sys_type": {"remote_field": "book"},
+      "sys_updated": {"remote_field": "prep_sys_updated"},
+      "sys_url_view": {"remote_field": "prep_book_url"},
+      "tags": {"remote_field": "prep_sys_tags"},
+      "thumbnail": {"remote_field": "field_thumbnail_url"},
+      "web_reader_link": {"remote_field": "field_web_reader_url"}
+    },
+    "preprocessors" : [
+      {
+        "name"     : "Remote id copy",
+        "class"    : "org.jboss.elasticsearch.tools.content.AddMultipleValuesPreprocessor",
+        "settings" : {
+          "prep_id_remote"                 :"{nid}",
+          "prep_id"                        :"jbossdeveloper_book-{nid}",
+          "prep_sys_title"                 :"{title}",
+          "prep_book_description"          :"{field_book_description}",
+          "prep_book_url"                  :"{field_book_url}",
+          "prep_categories"                :"{field_categories}",
+          "prep_sys_created"               :"{field_published_date}",
+          "prep_isbn"                      :"{field_isbn}",
+          "prep_publisher"                 :"{field_publisher}",
+          "prep_sys_content_content-type"  :"text/plain",
+          "prep_sys_content_provider"      :"jboss-developer",
+          "prep_sys_content_type"          :"jbossdeveloper_book",
+          "prep_sys_tags"                  :[],
+          "prep_pull_from_google"          :"{field_pull_from_google}"
+
+        }
+      },
+      {
+        "name"     : "Nullify false strings",
+        "class"    : "org.jboss.elasticsearch.tools.content.ScriptingPreprocessor",
+        "settings" : {
+          "script_engine_name"  : "JavaScript",
+          "script"              : "
+
+          var books_fields = ['field_published_date', 'field_thumbnail_url', 'field_web_reader_url', 'field_preview_url', 'field_categories', 'field_average_rating'];
+          for (var i=0; i < books_fields.length; i++){
+            var field_value = data.get(books_fields[i]);
+            if (field_value.toString() == 'false'){
+              data.put(books_fields[i], null);
+            }else{
+              data.put(books_fields[i], field_value);
+
+            }
+          }
+      "
+      }
+    },{
+        "name"     : "Format author array from string to array",
+        "class"    : "org.jboss.elasticsearch.tools.content.ScriptingPreprocessor",
+        "settings" : {
+          "script_engine_name"  : "JavaScript",
+          "script"              : "
+
+          var author_array = new Array();
+          var author = data.get('field_author').split(',');
+          for (var i=0; i < author.length; i++){
+        author_array[i] = author[i];
+    }
+      data.put('prep_sys_authors', author_array);
+      "
+      }
+    },{
+        "name"     : "String to ISO Date formatter",
```

Comment:
I think you've reimplemented in JS this preprocessor https://github.com/searchisko/structured-content-tools/blob/master/src/main/java/org/jboss/elasticsearch/tools/content/LongToTimestampValuePreprocessor.java